kong/plugins/ai-proxy/drivers/llama2.lua

```diff
@@ -0,0 +1,291 @@
+  ["llm/v1/chat"] = function(response_string, route_type)
+    local response_table, err = cjson.decode(response_string)
+    if err then
+      return nil, "failed to decode cohere response"
+    end
+    
+    if (not response_table) or (not response_table.data) or (#response_table.data > 1) then
+      return nil, "cannot parse response from llama2 endpoint"
+    elseif (not response_table.data[1].generated_text) then
+      return nil, "response data is empty from llama2 endpoint"
+    else
+      local split_response = split(response_table.data[1].generated_text, "[/INST]")
+      if not split_response or #split_response < 1 then
+        return nil, "response did not contain a system reply"
+      end
+  
+      local response_object
+
+      -- good
+      if route_type == "llm/v1/chat" then
+        response_object = {
+          choices = {
+            [1] = {
+              message = {
+                content = split_response[#split_response],
+                role = "assistant",
+              },
+              index = 0,
+            }
+          },
+          object = "chat.completion",
+        }
+  
+      elseif route_type == "llm/v1/completions" then
+        response_object = {
+          choices = {
+            [1] = {
+              index = 0,
+              text = split_response[#split_response],
+            }
+          },
+          object = "text_completion",
+        }
+  
+      end
+
+      return cjson.encode(response_object)
+    end
+  end,
+
+  ["llm/v1/completions"] = function(response_string, route_type)
+    local response_table, err = cjson.decode(response_string)
+    if err then
+      return nil, "failed to decode cohere response"
+    end
+    
+    if (not response_table) or (not response_table.data) or (#response_table.data > 1) then
+      return nil, "cannot parse response from llama2 endpoint"
+    elseif (not response_table.data[1].generated_text) then
+      return nil, "response data is empty from llama2 endpoint"
+    else
+      local split_response = split(response_table.data[1].generated_text, "[/INST]")
+      if not split_response or #split_response < 1 then
+        return nil, "response did not contain a system reply"
+      end
+  
+      local response_object
+
+      -- good
+      if route_type == "llm/v1/chat" then
+        response_object = {
+          choices = {
+            [1] = {
+              message = {
+                content = split_response[#split_response],
+                role = "assistant",
+              },
+              index = 0,
+            }
+          },
+          object = "chat.completion",
+        }
+  
+      elseif route_type == "llm/v1/completions" then
+        response_object = {
+          choices = {
+            [1] = {
+              index = 0,
+              text = split_response[#split_response],
+            }
+          },
+          object = "text_completion",
+        }
+  
+      end
+
+      return cjson.encode(response_object)
+    end
```

Comment:
Ignore this if I am mis-reading the control flow, but I think ending your first if/else statement early would help because it means we can reduce one level of indentation for the rest of the function (always a win in my book!).

```suggestion
    if (not response_table) or (not response_table.data) or (#response_table.data > 1) then
      return nil, "cannot parse response from llama2 endpoint"

    elseif (not response_table.data[1].generated_text) then
      return nil, "response data is empty from llama2 endpoint"
    end

    local split_response = split(response_table.data[1].generated_text, "[/INST]")
    if not split_response or #split_response < 1 then
      return nil, "response did not contain a system reply"
    end

    local response_object

    -- good
    if route_type == "llm/v1/chat" then
      response_object = {
        choices = {
          [1] = {
            message = {
              content = split_response[#split_response],
              role = "assistant",
            },
            index = 0,
          }
        },
        object = "chat.completion",
      }

    elseif route_type == "llm/v1/completions" then
      response_object = {
        choices = {
          [1] = {
            index = 0,
            text = split_response[#split_response],
          }
        },
        object = "text_completion",
      }

    end

    return cjson.encode(response_object)
```

Reply:
That's right, I fixed it,.